### ISSUE TITLE
not respecting -Wait or exit codes of powershell scripts

### ISSUE THREAD

**Issue by levilovelock** [masterzen/winrm#23]
If I have a script like:

```
Start-Process -FilePath "powershell.exe" -ArgumentList @("/c", "sleep", "5") -Wait
Exit 1
```

And run it under the powershell provisioner, I see two errors. Firstly, the wait is not respected and also the return code is 0, so it passes.

I get each of those behaviour when I invoke the command through the winrm binary . Not sure if you guys know if there's something I can add here to make it behave as I want?

I should note that this works for blocking the script:

```
$p = Start-Process -FilePath "powershell.exe" -ArgumentList @("/c", "sleep", "5") -Wait -PassThru
while ($p.HasExited -ne "True") {
  Start-Sleep -Milliseconds 500
}
```

Still can't work around the exit status though!

**Comment by hach-que** [comment]
Can you try running this under an Administrator PowerShell prompt on the remote host, and then running your command again?  I'd be interested to know if it solves the issue you're seeing.

```
# Prevent Powershell from using stupid defaults.
Set-Item -ErrorAction SilentlyContinue `
  WSMAN:localhost\Shell\MaxMemoryPerShellMB 100000000
Set-Item -ErrorAction SilentlyContinue `
  WSMAN:localhost\Shell\MaxShellsPerUser 10000

Push-Location WSMAN:localhost\Plugin\
foreach ($dir in Get-ChildItem) {
    $name = $dir.Name
    try {
        Set-Item -ErrorAction SilentlyContinue `
          -Path "$name\Quotas\MaxConcurrentCommandsPerShell" 100000000
    } catch [System.InvalidOperationException] {
    }
    try {
        Set-Item -ErrorAction SilentlyContinue `
          -Path "$name\Quotas\MaxMemoryPerShellMB" 10000
    } catch [System.InvalidOperationException] {
    }
}
Pop-Location
```

This increases the limits well beyond their defaults (and does so more extensively than the instructions in the README).  I was just about to file an issue to update the README with these instructions instead of the current ones.

I've noticed that on my build servers, before running those commands I'd sometimes get commands exiting or terminating early, with `winrm` having an exit code of 0, even though the commands were clearly interrupted in the middle of processing.  I believe it might be a case of the WinRM server on the Windows machine terminating the shell because it reaches it's maximum memory limit (or at least the processing hitting OOM).

**Comment by hach-que** [comment]
Upon further testing, it doesn't look like this solves the problem, it just allows commands to run for longer (for some strange reason).

**Comment by hach-que** [comment]
This should be fixed with PR #25.

**Comment by masterzen** [comment]
PR #25 has been merged-in, so closing. Reopen if it's not fixed.
